clear-intermediate-datasets/resource/get_projects_and_datasets.py

```diff
@@ -0,0 +1,29 @@
+import dataiku
+def do(payload, config, plugin_config, inputs):
+    """
+    Create list of options of projects in instance.
+    """
+    client = dataiku.api_client()
+    # Set project dropdown
+    if payload.get('parameterName') == "project_key":
+        projects = client.list_projects()
+        choices = []
+        for project in projects:
+            choices.append({"value": project.get('projectKey'), "label": project.get('projectKey')})
+        return {"choices": choices}
+    # Set dataset multiselect box - Display list of datasets of current project if no project has been selected
+    elif payload.get('parameterName') == "datasets_to_exclude" and config.get("project_key", None) == None:
+        project = client.get_default_project()
+        datasets = project.list_datasets()
+        choices = []
+        for dataset in datasets:
+            choices.append({"value": dataset.get('name'), "label": dataset.get('name')})
+        return {"choices": choices}
+    # Set dataset multiselect box - Display list of datasets if a project has been selected
+    elif payload.get('parameterName') == "datasets_to_exclude":
+        project = client.get_project(config.get("project_key"))
+        datasets = project.list_datasets()
+        choices = []
+        for dataset in datasets:
+            choices.append({"value": dataset.get('name'), "label": dataset.get('name')})
+        return {"choices": choices}    
```

Comment:
This part could be refactored:

1. It's better to not reevaluate the same condition: Instead of doing elif X and Y then 1 elif X then 2, there will be less comparisons doing elif X : if Y then 1 else 2
2. The content of each elif is very similar,  could be simplified with reusable functions/ variable declarations outside of the ifs instead

Reply:
Hi Jane, thanks for the feedback. Regarding 1. I initially tried your suggested approach but the dropdown was breaking when I would change projects. I'll give it another try.